src/qlever/commands/stop.py

```diff
@@ -65,6 +69,33 @@ def additional_arguments(self, subparser) -> None:
                                help="Do not look for containers, only for "
                                     "native processes")
 
+    @staticmethod
+    def stop_process_with_regex(cmdline_regex: str) -> list[bool] | None:
+        stop_process_results = []
+        for proc in psutil.process_iter():
+            try:
+                pinfo = proc.as_dict(
+                    attrs=[
+                        "pid",
+                        "username",
+                        "create_time",
+                        "memory_info",
+                        "cmdline",
+                    ]
+                )
+                cmdline = " ".join(pinfo["cmdline"])
+            except Exception as e:
+                log.debug(f"Error getting process info: {e}")
+                return None
+            if re.search(cmdline_regex, cmdline):
+                log.info(
+                    f"Found process {pinfo['pid']} from user "
+                    f"{pinfo['username']} with command line: {cmdline}"
+                )
+                log.info("")
+                stop_process_results.append(stop_process(proc, pinfo))
+        return stop_process_results
```

Comment:
Looks like this should be  in `utils.py` (with a proper comment explaining what is does)